### PR TITLE
DEPR: remove deprecated freqs/abbrevs 'A', 'A-DEC', 'A-JAN', etc.

### PR DESCRIPTION
**Pull request by natmokval**
xref #55252

remove deprecated string 'A', 'A-DEC', etc. denoting timedelta abbreviations and timeseries frequencies

### CODE REVIEW

File: pandas/_libs/tslibs/dtypes.pxd [hunk 1]
```diff
@@ -18,6 +18,7 @@ cdef dict c_DEPR_ABBREVS
 cdef dict attrname_to_abbrevs
 cdef dict npy_unit_to_attrname
 cdef dict attrname_to_npy_unit
+cdef set c_REMOVED_ABBREVS
```
Comment (WillAyd) [inline pandas/_libs/tslibs/dtypes.pxd:21]: I don't think we need to keep track of these over time. Should just be able to remove from `c_DEPR_ABBREVS` and call it a day I think

File: pandas/tests/frame/methods/test_asfreq.py [hunk 2]
```diff
@@ -283,3 +281,19 @@ def test_asfreq_unsupported_freq(self, freq, error_msg):
...
+        ],
+    )
+    def test_asfreq_frequency_A_raisees(self, freq, freq_depr):
+        msg = f"Invalid frequency: {freq_depr[1:]}"
+
+        index = date_range("1/1/2000", periods=4, freq=f"{freq[1:]}")
+        df = DataFrame({"s": Series([0.0, 1.0, 2.0, 3.0], index=index)})
+
+        with pytest.raises(ValueError, match=msg):
+            df.asfreq(freq=freq_depr)
```
Comment (MarcoGorelli) [inline pandas/tests/frame/methods/test_asfreq.py:299]: it's not clear to me what you're doing with `freq` in this test - I think `index` can just be instatiated with `'YE'` in both cases?

Comment (natmokval) [reply pandas/tests/frame/methods/test_asfreq.py]: yes, you are right, it's sufficient to check only `freq="A"`. I simplified the test and added a note to `v3.0.0`.
Could you please take a look at my changes?

**Review by MarcoGorelli — APPROVED** [review]
looks good to me

good to explicitly check that these raise here, if there's something I've learned about all these offsets is that anything could happen 😄

**Comment by natmokval** [comment]
Thanks @MarcoGorelli for reviewing this PR.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @natmokval

**Pull request merged**